core/jvm/src/main/scala/zio/internal/FiberInbox.scala

```diff
@@ -0,0 +1,15 @@
+package zio.internal
+
+import org.jctools.queues.atomic.unpadded.MpscLinkedAtomicUnpaddedQueue
```

Comment:
Shouldn't we use `import org.jctools.queues.unpadded.MpscLinkedUnpaddedQueue` instead?

Reply:
So their class hierarchy can look a bit messy due to their codegen, but the difference is the `atomic` variant of the queue I've chosen uses atomic references rather than memory barriers for its nodes which performs better under all scenarios in our benchmarks.

Reply:
What is the difference you're seeing in the benchmarks and in which ones?

Reply:
@kyri-petrou it takes about 2 hours to run the benchmark suite each time, so 4 hours to do a mostly fair comparison (I have an idle machine I'm using to run the benchmarks to avoid as much inconsistency as possible, but I try to still run them close together).

One that very clearly demonstrated the difference between atomics and barriers is `DeepAsyncBenchmark.zioDeepAsyncOne`, I can get you a fresh run of that quite quickly and I'll reply with the numbers in this thread.

If there are others that you specifically would like to compare atomic vs barrier please let me know.

Reply:
That's OK, I think we can keep the atomic-based one for the time being and revisit this in the future if we want to